### PR TITLE
Remove incorrect links to zdevito/ATen

### PR DESCRIPTION
**Pull request by samestep**
Similar to #49028, this PR removes a few more references to https://github.com/zdevito/ATen.

- The links for Functions.h, Tensor.h, and Type.h are simply broken, probably because they refer to `master` rather than a specific commit (cf. #47066)
- I'm unsure about the change to the `about` section of `aten/conda/meta.yaml`; can someone comment on whether I am understanding that field correctly?
- The reference to https://github.com/zdevito/ATen/issues/163 remains [in `tools/autograd/derivatives.yaml`](https://github.com/pytorch/pytorch/blob/cd608fe59b70fa7cafb07110096b2e023a8b6e9c/tools/autograd/derivatives.yaml#L91), because the contents of that issue discussion don't seem to be mirrored anywhere else.

### CODE REVIEW

**Review by walterddr — APPROVED** [review]
Looks good. But should we add links to pytorch/pytorch instead of just removing them?

**Comment by samestep** [comment]
> Looks good. But should we add links to pytorch/pytorch instead of just removing them?

Definitely, if you know what those links would be :) I wasn't able to find the corresponding files, but if someone knows where they are then I'll add them in this PR

**Comment by samestep** [comment]
(this is why I think it would be nice to have a lint that warns about links to files on GitHub but don't refer to a specific commit)

**Comment by samestep** [comment]
@walterddr should I wait a bit longer in case someone knows what to replace the links with, or should I just merge this PR?

File: caffe2/contrib/aten/README.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 that makes these functions available in Caffe2 as an operator. It also makes it accessible using the
 ToffeeIR.
 
 
 ### Example Usage in Caffe2
 
-First identify a function in ATen you want to call in [Functions.h](https://github.com/zdevito/ATen/blob/master/doc/Functions.h),
-[Tensor.h](https://github.com/zdevito/ATen/blob/master/doc/Tensor.h), or [Type.h](https://github.com/zdevito/ATen/blob/master/doc/Type.h).
+First identify a function in ATen you want to call in Functions.h,
+Tensor.h, or Type.h.
```
Comment (walterddr) [inline caffe2/contrib/aten/README.md:12]: i dont even think this is still valid. cant find the link

**Review by walterddr — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Comment by samestep** [comment]
I think the land failed somehow, trying again now

**Pull request closed without merging**